mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ConfigRefDeclarationEnricher.java

```diff
@@ -58,7 +63,16 @@ public void enrich(ExtensionLoadingContext extensionLoadingContext) {
     final ExtensionDeclaration declaration = extensionLoadingContext.getExtensionDeclarer().getDeclaration();
     Multimap<ComponentDeclaration, ConfigurationDeclaration> componentsConfigs = getComponentConfigsMap(declaration);
     componentsConfigs.asMap()
-        .forEach((component, configs) -> component.getDefaultParameterGroup().addParameter(buildConfigRefParameter(configs)));
+        .forEach((component, configs) -> {
+          if (component.getDefaultParameterGroup().getParameters().stream()
```

Comment:
if this is because of Apikit, then this if should be predicated on some system property only set by Apikit. Otherwise any extension gets to define this, potentially in the wrong way.

That system property should not be part of the API and be clearly documented as *DO NOT USE*